File: nashorn/src/org/jetbrains/dukat/nashorn/DocumentCache.java
package org.jetbrains.dukat.nashorn;

import java.util.HashMap;
import java.util.Map;

public class DocumentCache {
    private Map<String, Object> myDocumentMap = new HashMap<>();

    public void setDocument(String key, String path, Object sourceFile) {
        myDocumentMap.put(path, sourceFile);
    }

    public Object getDocument(String key, String path) {
        return myDocumentMap.get(path);
    }
}

File: interop-graal/src/org/jetbrains/dukat/interop/graal/DocumentCache.kt
package org.jetbrains.dukat.interop.graal

import java.util.HashMap

class DocumentCache {
    private val myDocumentMap = HashMap<String, Any>()

    fun setDocument(key: String, path: String, sourceFile: Any) {
        myDocumentMap[path] = sourceFile
    }

    fun getDocument(key: String, path: String): Any? {
        return myDocumentMap[path]
    }
}
